pkg/reconcilers/targetref_reconciler.go

```diff
@@ -94,6 +94,35 @@ func (r *TargetRefReconciler) FetchValidTargetRef(ctx context.Context, targetRef
 	return nil, fmt.Errorf("FetchValidTargetRef: targetRef (%v) to unknown network resource", targetRef)
 }
 
+// FetchAcceptedGatewayHTTPRoutes returns the list of HTTPRoutes that have been accepted as children of a gateway.
+func (r *TargetRefReconciler) FetchAcceptedGatewayHTTPRoutes(ctx context.Context, gwKey client.ObjectKey) (routes []gatewayapiv1beta1.HTTPRoute) {
+	logger, _ := logr.FromContext(ctx)
+	logger = logger.WithName("FetchAcceptedGatewayHTTPRoutes").WithValues("gateway", gwKey)
+
+	routeList := &gatewayapiv1beta1.HTTPRouteList{}
+	err := r.Client().List(ctx, routeList)
```

Comment:
Without any criteria, this could be over-killing. Can't we use Gateway status or the annotations to get this list?

Reply:
IIRC, common options for filtering results here are:
- labels
- resource fields ([not all fields are supported](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#supported-fields))
- namespace

I didn't know we could use annotations.

If the HTTPRoute allows us to use `parentRef` and/or `status.parents` for filtering, that would be great. I'll look into it.

Reply:
`s/annotations/labels/g`

Indeed, I meant labels. Anyway, enforcing users to add some kuadrant labels to the HTTPRoute resources only the controller can query the cluster is not very cool.